app/views/transactions/edit.js.erb

```diff
@@ -1,4 +1,20 @@
-if($("#<%= dom_id(@transaction, 'edit_row') %>").length == 0) {
-  $("<%=j render 'transactions/edit_transaction', transaction: @transaction %>").hide().insertAfter(".transaction#<%= dom_id(@transaction) %>").slideDown();
-  addTransactionFormMasks();
-}
+$('#modal_title').html("<i class='glyphicon glyphicon-pencil'></i> Edit Transaction");
+
+<% if @transaction.category.type == 'Income' %>
+  $('#income').html("<%= j render('transactions/form', type: nil, transaction: @transaction) %>");
```

Comment:
@yunixon in edit modal show content only for one tab (type of transaction). Don't show over tabs.